src/main/java/Duke.java

```diff
@@ -1,10 +1,203 @@
+        boolean isEvent = words[0].equals("event");
+        boolean isDelete = words[0].equals("delete");
+        boolean invalidCommand = !(isList || isDone || isTodo || isDeadline || isEvent || isDelete);
+        if(invalidCommand) {
+            throw new DukeException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+        }
+    }
+
+    public static void validateToDoInput(String[] words) throws DukeException {
+        boolean invalidToDoInput = words.length == 1;
+        if(invalidToDoInput) {
+            throw new DukeException("☹ OOPS!!! The description of a todo cannot be empty.");
+        }
+    }
+
+    public static void validateListInput(String[] words) throws DukeException {
+        boolean invalidListInput = words.length > 1;
+        if (invalidListInput) {
+            throw new DukeException("☹ OOPS!!! The description of a list should be empty.");
+        }
+    }
+
+    public static void loadFile(ArrayList<Task> tasks) throws FileNotFoundException {
+        File f = new File("tasks.txt");
+        if (!f.exists()) {
+            throw new FileNotFoundException();
+        }
+        Scanner s = new Scanner(f);
+        while (s.hasNext()) {
+            String input = s.nextLine();
+            String[] words = input.split(" ");
+            boolean isTodo = words[0].equals("T");
+            boolean isDeadline = words[0].equals("D");
+            boolean isEvent = words[0].equals("E");
+            boolean isDone = words[1].equals("Y");
+            input = input.substring(4);
+            words = input.split("/d");
+            if (isDeadline) {
+                Deadline deadline = new Deadline(words[0], words[1]);
+                tasks.add(deadline);
+            }
+            else if (isEvent) {
+                Event event = new Event(words[0], words[1]);
+                tasks.add(event);
+            }
+            else if (isTodo) {
+                ToDo toDo = new ToDo(words[0]);
+                tasks.add(toDo);
+            }
+            if (isDone) {
+                tasks.get(tasks.size()-1).taskComplete();
+            }
+        }
+    }
+
+    public static void saveFile(String filepath, ArrayList<Task> tasks) throws IOException {
+        FileWriter fw = new FileWriter(filepath);
+        for (Task task: tasks) {
+            fw.write(task.toSaveFormat());
+        }
+        fw.close();
+    }
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
-    }
-}
+        ArrayList<Task> Tasks =  new ArrayList<Task>();
+        String line;
+        String INTRO_MESSAGE = " Hello! I'm Duke\n" + " What can I do for you?";
+        String OUTRO_MESSAGE = "Bye. Hope to see you again soon!";
```

Comment:
Maybe can store this above as static variables instead of always reinstantiating in main